agent/api/statechange.go

```diff
@@ -481,7 +481,7 @@ func (AttachmentStateChange) GetEventType() statechange.EventType {
 	return statechange.AttachmentEvent
 }
 
-func buildManagedAgentStateChangePayload(change ManagedAgentStateChange) *ecsmodel.ManagedAgentStateChange {
+func buildManagedAgentStateChangePayload(change ManagedAgentStateChange) *types.ManagedAgentStateChange {
```

Comment:
Is there any specific reason why we don't just return type `types.ManagedAgentStateChange` here? Returning pointer type `*types.ManagedAgentStateChange`  here seems unnecessary to me (i.e., we just end up dereferencing on L427 anyway), but I may be missing something.

Furthermore, it seems like in most other places in this pull request we replace `*ecsmodel.<INSERT TYPE HERE>` with `types.<INSERT TYPE HERE>`  as opposed to `*types.<INSERT TYPE HERE>`.

Reply:
From a related thread ([ref](https://github.com/aws/amazon-ecs-agent/pull/4447#discussion_r1951671163)) - the reasoning is so that existing callers may do nil checks.